module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -75,6 +117,25 @@ public function findResource(
         $query->setParameters($parameters);
         $result = $query->getResult();
 
+        if (empty($result) && $create) {
+            $resource = $this->createEntity()
+                ->setRecordId($id)
+                ->setSource($source);
+
+            // Load record if it was not provided:
+            if (null === $driver) {
```

Comment:
I think this if statement can be simplified to: `$driver ??= $this->recordLoader->load($id, $source);`